pkg/platform/abstract/platform.go

```diff
@@ -216,23 +232,41 @@ func (ap *Platform) ResolveDefaultNamespace(defaultNamespace string) string {
 	return ""
 }
 
-func (ap *Platform) functionBuildRequired(createFunctionOptions *platform.CreateFunctionOptions) (bool, error) {
+func (ap *Platform) functionBuildRequired(existingFunctionConfig *functionconfig.ConfigWithStatus,
+	createFunctionOptions *platform.CreateFunctionOptions) (bool, error) {
+
+	// check if we have something to compare to, if so, check if anything changed
```

Comment:
If at first deploy the function failed, it is created and has `sourceHash` value of empty string.
So a check needs to be added otherwise `functionBuildRequired` will return false on failed functions.

The check can be to see that `sourceHash` is not empty if you think it's clearer